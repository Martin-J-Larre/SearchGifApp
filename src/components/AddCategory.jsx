import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;

    setCategories((cat) => [inputValue, ...cat]);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Search"
        value={inputValue}
        // onChange={(e) => onInputChange(e)}
        onChange={onInputChange}
      />
    </form>
  );
};
