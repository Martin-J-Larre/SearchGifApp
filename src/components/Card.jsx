import PropTypes from "prop-types";

export const Card = ({ title, url, id }) => {
  return (
    <>
      <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
      </div>
    </>
  );
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
