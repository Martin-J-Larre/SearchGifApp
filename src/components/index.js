export * from "./AddCategory";
export * from "./Card";
export * from "./GifGrid";
