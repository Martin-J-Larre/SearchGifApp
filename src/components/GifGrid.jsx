import { Card } from "./Card";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div>
      <h3>{category}</h3>
      {isLoading && <h4>Loading...</h4>}
      <div className="card-grid">
        {images.map((img) => (
          <Card key={img.id} {...img} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
