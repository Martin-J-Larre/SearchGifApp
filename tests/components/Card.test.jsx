import { render, screen } from "@testing-library/react";
import { Card } from "../../src/components/Card";

describe("Test in Card compenent", () => {
  const title = "String";
  const url = "https://string-string/";

  test("should match with snapshot", () => {
    const { container } = render(<Card title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show image with URL and Alt text", () => {
    render(<Card title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("should show a title in the component", () => {
    render(<Card title={title} url={url} />);
    expect(screen.getByAltText(title)).toBeTruthy();
  });
});
