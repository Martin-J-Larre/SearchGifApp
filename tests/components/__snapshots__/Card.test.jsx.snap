// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Card compenent should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="String"
      src="https://string-string/"
    />
    <p>
      String
    </p>
  </div>
</div>
`;
